com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/TextureUtil.java

```diff
@@ -267,13 +266,50 @@ public static TextureImage createCombinedTextureImage(TextureImage[] textures, T
         return combinedImageBuilder.build();
     }
 
+    public static BufferedImage Resize(BufferedImage img, int newW, int newH, int type) {
+        Image tmp = img.getScaledInstance(newW, newH, Image.SCALE_SMOOTH);
+        BufferedImage dimg = new BufferedImage(newW, newH, type);
+
+        Graphics2D g2d = dimg.createGraphics();
+        g2d.drawImage(tmp, 0, 0, null);
+        g2d.dispose();
+
+        return dimg;
+    }
+
     // Public api
     public static TextureImage createMultiPageTexture(List<BufferedImage> images, TextureImage.Type textureType, TextureProfile texProfile, boolean compress) throws TextureGeneratorException {
         TextureImage textureImages[] = new TextureImage[images.size()];
+
+        // Since external tools may provide pages of verying sizes, we need to make sure
+        // our output is of the correct dimensions (as we're using a texture array)
+        int maxWidth = 0;
+        int maxHeight = 0;
+        int firstType = -1;
+
+        for (int i = 0; i < images.size(); i++)
+        {
+            BufferedImage image = images.get(i);
+            maxWidth = Math.max(maxWidth, image.getWidth());
+            maxHeight = Math.max(maxHeight, image.getHeight());
```

Comment:
There are different strategies to choose I guess, but think that we should create the biggest texture, as it's passed as the data.